tests/components/test_snips.py

```diff
@@ -1,45 +1,227 @@
-"""Test the Snips component."""
-import asyncio
-
-from homeassistant.bootstrap import async_setup_component
-from tests.common import async_fire_mqtt_message, async_mock_intent
-
-EXAMPLE_MSG = """
-{
-    "input": "turn the lights green",
-    "intent": {
-        "intentName": "Lights",
-        "probability": 1
-    },
-    "slots": [
+"""The tests for the Dialogflow component."""
+import unittest
+
+from homeassistant.core import callback
+from homeassistant.components import snips
+from homeassistant.setup import async_setup_component, setup_component
+from tests.common import (get_test_home_assistant, fire_mqtt_message,
+                          mock_mqtt_component, mock_component, mock_service)
+
+
+class TestSnips(unittest.TestCase):
+    """Test Snips."""
+
+    def setUp(self):
+        """Setup things to be run when tests are started."""
+        self.hass = get_test_home_assistant()
+        self.calls = []
+
+        mock_mqtt_component(self.hass)
+        assert async_setup_component(self.hass, snips.DOMAIN, {
+            "snips": {},
+        })
+        self.hass.start()
+
+    def tearDown(self):  # pylint: disable=invalid-name
+        """Stop everything that was started."""
+        self.hass.stop()
+
+    @callback
+    def record_calls(self, *args):
+        """Helper for recording calls."""
+        self.calls.append(args)
+
+    def test_intent_call_service(self):
+        """Test intent calling a service."""
+        mock_component(self.hass, 'mqtt')
```

Comment:
No need for this, `setUp` is called before each test.

Reply:
That's what I thought but it wouldn't work when only called in setUp. Let me give it a better poke tomorrow. I am sure it's a matter of scoping but maybe a refactor to get away from unittest.TestCase will resolve that.

Reply:
Fixed this at least